pkg/controller/cache/claim.go

```diff
@@ -112,11 +112,6 @@ func ConfigureReplicationGroup(_ context.Context, cm resource.Claim, cs resource
 
 	rg.Spec = *spec
 
-	// NOTE(muvaf): ReplicationGroup actually supports memcached as well but the
-	// claim is a _Redis_Cluster, so, we override the value no matter what. Users
-	// should be able to get a memcached through manually creating the ReplicationGroup
-	rg.Spec.ForProvider.Engine = v1beta1.CacheEngineRedis
-
```

Comment:
What's the context around this removal?

Reply:
`Engine` is a required field and if it's not supplied, the creation of the resource class is rejected immediately anyway. So, for engine type, we rely completely on what resource class gives us and only that, which could be either `redis` or `memcached`.